django/core/management/commands/makemessages.py

```diff
@@ -548,9 +553,10 @@ def find_files(self, root):
                         self.stdout.write("ignoring directory %s" % dirname)
                 elif dirname == "locale":
                     dirnames.remove(dirname)
-                    self.locale_paths.insert(
-                        0, os.path.join(os.path.abspath(dirpath), dirname)
-                    )
+                    locale_dir = os.path.join(os.path.abspath(dirpath), dirname)
+                    if locale_dir in self.locale_paths:
+                        self.locale_paths.remove(locale_dir)
+                    self.locale_paths.insert(0, locale_dir)
```

Comment:
Hah, I had a go at fixing this myself this week before looking at the Django ticket tracker and came up with almost identical code:
```python
locale_dir_path = os.path.join(os.path.abspath(dirpath), dirname)
if locale_dir_path not in self.locale_paths:
    self.locale_paths.insert(
        0, locale_dir_path
    )
```

Reply:
I have added the test that should detect errors with locale directory assignment for given source files (`test_correct_translatable_file_locale_dir`). The 3rd subtest, where app locale directory is explicitly set in `LOCALE_PATHS` is the one that passes with current implementation, and fails with suggested change.